### PR TITLE
Add huggingface's link to external projects

### PR DESCRIPTION
**Pull request by knshnb**
## Motivation

Optuna is used in `Trainer.hyperparameter_search` in Hugging Face.

## Description of the changes

Added Hugging Face's link to external projects in README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -113,6 +113,7 @@ In addition, integration modules are available for the following libraries, prov
 
 ### External projects using Optuna
 
+* [Hugging Face Trainer's Hyperparameter Search](https://huggingface.co/docs/transformers/main/main_classes/trainer#transformers.Trainer.hyperparameter_search)
```
Comment (toshihikoyanase) [inline README.md:116]: Not a strong opinion, but how about stop using the title case like other entries?
```suggestion
* [Hugging Face Trainer's hyperparameter search](https://huggingface.co/docs/transformers/main/main_classes/trainer#transformers.Trainer.hyperparameter_search)
```

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. I was not aware of huggingface's use case.
I have a minor comment.

File: README.md (hunk 1 above)
Comment (knshnb) [reply README.md]: Thanks, that makes sense.

**Review by toshihikoyanase — APPROVED** [review]
Thank you for your update. LGTM!

**Pull request merged**